backend/src/main/java/com/jobwait/domain/Answer.java

```diff
@@ -1,36 +1,69 @@
-                                "jobApplicationCount" }),
-                @JsonSubTypes.Type(value = ValidEducationLevelAnswer.class, name = "educationLevel"),
-                @JsonSubTypes.Type(value = ValidWorkModelAnswer.class, name = "workModel"),
-                @JsonSubTypes.Type(value = ValidWorkContractAnswer.class, name = "workContract"),
-                @JsonSubTypes.Type(value = LocalDateAnswer.class, names = { "jobAcceptDate",
-                                "jobSearchStartDate" }),
-})
-public interface Answer<T> {
-        public T getValue();
-
-        public void setValue(T value);
-
-        public String getType();
-
-        public void setType(String type);
-
-        public void setSQLStatement(PreparedStatement statement, Integer statementIndex) throws SQLException;
+import java.time.LocalDate;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.jobwait.jackson.AnswerDeserializer;
+
+@JsonDeserialize(using = AnswerDeserializer.class)
+public class Answer {
+    public String questionKey;
+    public AnswerType answerType;
+    public Object answerValue;
+
+    public Answer(String questionKey, AnswerType answerType, Object answerValue) {
+        this.questionKey = questionKey;
+        this.answerType = answerType;
+        this.answerValue = answerValue;
+    }
+
+    public String getQuestionKey() {
+        return this.questionKey;
+    }
+
+    public AnswerType getAnswerType() {
+        return this.answerType;
+    }
+
+    public Object getAnswerValue() {
+        switch (this.getAnswerType()) {
+            case AnswerType.STRING:
+                return this.answerValueAsString();
+            case AnswerType.INTEGER:
+                return this.answerValueAsInteger();
+            case AnswerType.ENUM:
+                return this.answerValueAsString();
+            case AnswerType.DATE:
+                return this.answerValueAsDate();
+            default:
+                throw new RuntimeException("Answer type not defined (impossible error)");
+        }
+    }
+
+    public String answerValueAsString() {
+        return (String) this.answerValue;
+    }
+
+    public Integer answerValueAsInteger() {
+        return (Integer) this.answerValue;
+    }
+
+    public LocalDate answerValueAsDate() {
+        return (LocalDate) this.answerValue;
+    }
+
+    public static boolean assertValidAnswerType(AnswerType answerType, Object answerValue) {
+        switch (answerType) {
+            case AnswerType.STRING:
+                return answerValue.getClass().equals(String.class);
+            case AnswerType.INTEGER:
+                return answerValue.getClass().equals(Integer.class);
+            case AnswerType.ENUM:
+                return answerValue.getClass().equals(String.class);
+            case AnswerType.DATE:
+                return answerValue.getClass().equals(LocalDate.class);
+            default:
+                return false;
```

Comment:
i understand the argument for this, my question is why did i waste time writing up the separate classes which we distinctly spoke about being a good idea for separation only for it to all be rewritten. i dont much care about how its done but its a massive waste of my time. why should i write code if in one day you will decide to remove it all on a whim? i basically had this switch made in my very first draft of the Answers PR and we spoke about removing it

Reply:
Yes, this is the most controversial of my changes, read my other comment on reducing boilerplate. I'll respond to the overall theme of rewriting code after I finish responding to each comment